vm/vm.go

```diff
@@ -288,6 +288,69 @@ func (vm *Engine) validateStateTransition(virtualTx *wire.MsgTx) error {
 				spew.Sdump(witness.PrevID))
 		}
 
+		// Assume the default flag of SigHashAll.
+		sighashFlag := txscript.SigHashAll
+
+		const (
+			leafVersionEven = byte(txscript.BaseLeafVersion)
+			leafVersionOdd  = byte(txscript.BaseLeafVersion | 1)
+		)
+
+		for _, txWitness := range witness.TxWitness {
```

Comment:
I think we can pull this out into a helper function, and even just re-use `txscript` more. 

But even then, each signature within a control block can also have its own sighash flag. Thinking about it more, I don't think we need a new tx version? In that now we can sign it, but the Bitcoin layer is what would handle all the verification.

This PR is then more about the RPC level stuff, and the itest to confirm the case.

Reply:
Yes I don't believe we need a script version bump.